darts/models/tcn_model.py

```diff
@@ -155,7 +152,7 @@ def __init__(self,
 
         # If num_layers is not passed, compute number of layers needed for full history coverage
         if num_layers is None and dilation_base > 1:
-            num_layers = math.ceil(math.log((input_length - 1) * (dilation_base - 1) / (kernel_size - 1) / 2,
+            num_layers = math.ceil(math.log((input_length - 1) * (dilation_base - 1) / (kernel_size - 1) / 2 + 1,
```

Comment:
Because without it the formula is wrong. I figured this out while re-reading my TCN article, one of the derivations I made was wrong, now I'm very certain it's correct! Now there is also a test case that tests the formula.